### ISSUE TITLE
Generic alias methods with type constraints fail compilation

### ISSUE THREAD

**Issue by kcamp** [cake-build/cake#1566]
### What You Are Seeing?
Creating a generic alias method that contains type constraints fails compilation.
The generated method signature fails to enforce the constraints, making the implementation invalid.

For example:
```
public static TResult ReadResults<TResult >(this ICakeContext context, IResultReader<TResult> reader) 
where TResult : class      
```

The script generated by the engine results in this, roughly:
```
public TResult ReadResults<TResult>(IResultReader<TResult > reader)
{
    // invalid invocation because the generated method signature doesn't enforce the constraint
    return The.Namespace.ReadResults<TResult>(Context, reader);
}
```

### What is Expected?
Expect a valid alias method to generate script and compile

### What version of Cake are you using?
0.19.2

### Are you running on a 32 or 64 bit system?
64

### What environment are you running on?  Windows? Linux? Mac?
Windows

### Are you running on a CI Server?  If so, which one?
No

### How Did You Get This To Happen? (Steps to Reproduce)

Write an generic alias method with type parameter constraints in place in a Cake add-in library.

### Output Log
```
Compiling build script...
Error: Roslyn.Compilers.CompilationErrorException: (2464,133): error CS0452: The type 'TResult' must be a reference type in order to use it as parameter 'TResult' in the generic type or method 'Apprenda.Cake.Artifacts.IArtifactManifestReader<TResult>'
   at Roslyn.Scripting.CommonScriptEngine.CompilationError(DiagnosticBag localDiagnostics, DiagnosticBag diagnostics)
   at Roslyn.Scripting.CommonScriptEngine.Compile(String code, String path, DiagnosticBag diagnostics, Session session, Type delegateType, Type returnType, CancellationToken cancellationToken, Boolean isInteractive, Boolean isExecute, CommonCompilation& compilation, Delegate& factory)
   at Roslyn.Scripting.CommonScriptEngine.Execute[T](String code, String path, DiagnosticBag diagnostics, Session session, Boolean isInteractive)
   at Roslyn.Scripting.Session.Execute(String code)
   at Cake.Scripting.Roslyn.Stable.DefaultRoslynScriptSession.Execute(Script script)
   at Cake.Core.Scripting.ScriptRunner.Run(IScriptHost host, FilePath scriptPath, IDictionary`2 arguments)
   at Cake.Commands.BuildCommand.Execute(CakeOptions options)
   at Cake.CakeApplication.Run(CakeOptions options)
   at Cake.Program.Main()
```

**Comment by kcamp** [comment]
I have a fix for this issue, will submit a PR shortly.

**Comment by devlead** [comment]
Fixed by #1567
